ElectronClient/generateSha512.js

```diff
@@ -0,0 +1,40 @@
+const fs = require('fs');
+const path = require('path');
+const sha512 = require('js-sha512');
+
+const generateChecksumFile = () => {
+	const distDirName = 'dist';
+	const distPath = path.join(__dirname, distDirName);
+	try {
```

Comment:
I think the logic should go like that:

- First check if we're on Linux - if not, exit now
- Secondly check if the .AppImage can be found - if not, throw an exception

Then proceed with your existing logic.

Reply:
I have the changed the flow to what you have instructed